README.md

```diff
@@ -316,11 +314,97 @@ it('should use custom element', async() => {
+
+For example, to run all tests in the default `'test'` group in Firefox and with a `3s` timeout:
+
+```bash
+d2l-test-runner --firefox --timeout=3000
+```
+
+A `d2l-test-runner.config.js` file in the current working directory (or the path from the `config` CLI flag) can also be used to configure `d2l-test-runner`.
+
+```javascript
+export default {
+  firefox: true,
+  timeout: 3000
+};
+```
+
+### Test Groups
+
+Tests are organized into groups, which can be configured and run together.
+
+The group name appears in the default `files` pattern (`'./test/**/*.<group>.js'`), making it typical for test files to have the group name as part of their extension. For example, the default group is `'test'` so all test files named `*.test.js` will belong to it by default. Similarly, the `vdiff` group contains files named `*.vdiff.js`.
+
+To run tests which match the pattern `'./test/**/*.mygroup.js'`:
+```bash
+d2l-test-runner --group mygroup
+```
+
+### Running a Subset of Tests
+
+While writing or debugging tests, it can be desirable to focus the runner on a subset of tests.
+
+### By File Name
+
+Use the `filter` option to filter by file name. It replaces any wildcards in the file name portion of the `files` pattern with the provided [glob](https://en.wikipedia.org/wiki/Glob_(programming)).
+
+For example, with the `'test'` group and default pattern `'./test/**/*.<group>.js'`, passing `d2l-test-runner --filter foo` will run tests which match `'./test/**/foo.test.js'`.
+
+Wildcards can still be used but need to be escaped. So `d2l-test-runner --filter foo\*` will run tests which match `'./test/**/foo*.test.js'`.
+
+### By Test Name
+
+Use the `grep` option to filter by test name. Only tests whose names match the provided string or regular expression will be run, regardless of file name.
+
+For example, `d2l-test-runner --grep foo` will run any test whose test suite(s) or name contains "foo".
+
+> **Note:** unfortunately, tests which do not match the grep value will be reported as failed instead of skipped.
+
+### Debugging Tests
+
+When tests don't go as expected, the next step is usually to debug them using the browser's built-in developer tools.
+
+There are two options for debugging:
+* `watch`: after running `d2l-test-runner --watch`, choose "D" (debug in the browser) and select which test file to launch
+* `open`: is a shortcut for `watch` that opens a browser window for each file sequentially. For large projects, use it in combination with `filter` to limit which file(s) are opened. For example: `d2l-test-runner --filter foo --watch`.
+
+With both `watch` and `open`, the browser debugger will be paused at the top of the file under test, which provides an opportunity to attach breakpoints if desired. A toolbar at the top of the screen then allows for each individual test to be skipped or run, as well as a "run all" option.
```

Comment:
```suggestion
With both `watch` and `open`, you can open the dev tools before starting the test run and the browser debugger will be paused at the top of the file under test. This provides an opportunity to attach breakpoints if desired. A toolbar at the top of the screen then allows for each individual test to be skipped or run, as well as a "run all" option.
```

Or is this only a Windows thing?

Reply:
Oh interesting... so for me dev-tools is always opened for me.

Reply:
I wonder if it's something I need to set?  That would be nice, it's a huge pain to open it and then when you start the screen resizes to small again

Reply:
Yeah so Chromium is the only one that consistently has the debugger window open for me. I'll adjust the wording to what you provided to cover both cases.